sdk/openai/Azure.AI.OpenAI/src/Custom/AzureCognitiveSearchChatExtensionConfiguration.Serialization.cs

```diff
@@ -17,17 +17,18 @@ void IUtf8JsonSerializable.Write(Utf8JsonWriter writer)
             writer.WritePropertyName("type"u8);
             writer.WriteStringValue(Type.ToString());
 
-            // Custom code note: everything *except* type goes into 'parameters'
+            // CUSTOM CODE NOTE: Everything *except* 'type' goes into 'parameters'
             writer.WriteStartObject("parameters"u8);
```

Comment:
I'm curious why we need this customization. Why not just amend typespec?

Reply:
This is to make the class easier to use: By hiding the "parameters" property bag and effectively flattening the class.